src/Mod/Sketcher/App/SketchObject.h

```diff
@@ -231,7 +246,7 @@ class SketcherExport SketchObject: public Part::Part2DObject
         return ExternalGeo.getValues();
     }
     /// rebuilds external geometry (projection onto the sketch plane)
-    void rebuildExternalGeometry(bool defining = false, bool intersection = false);
+    void rebuildExternalGeometry(std::optional<ExternalToAdd> extToAdd = std::nullopt);

```

Comment:
The thing is that this function rebuildExternalGeometry is used for both updating the external geos, and also to add new external geos.
I made like this to have minimal changes and such that it is actually much clearer than it was.
Why do you say that it's strange? Isn't optional exactly intended to be used like this?

Reply:
No, `std::optional` is mostly intended as return value to indicate via type system that return value might not be there. This is type safe alternative to returning nullptr. 

Another example is using `std::optional` as field to denote that it might be not set - yet again it is used as alternative to keeping nullptr or even to introduce difference between lack of value or zero value.

Using `std::optional` as argument is of course possible and has some uses, but generally indicates a problem with design, i.e. that function does two unrelated things. In such case you clearly indicated that flaw.

Reply:
ok